### PR TITLE
feat: server hardening

### PR DESCRIPTION
**Pull request by Bidon15**
## Overview
I found some small inconveniences that requires treatment. For example, we have outdated docs for daserver, no protection on PUT size on blobs. 

Should be reviewed only after #49 is merged

### CODE REVIEW

**Review by tuxcanfly — COMMENTED** [review]
Reminder to self to harden cache and fallback mechanism.

File: cmd/daserver/config.go [hunk 1]
```diff
@@ -196,14 +230,72 @@ func (c *Config) GetSubmissionTimeout() (time.Duration, error) {
...
+				return 0, fmt.Errorf("invalid number in byte size: %w", err)
+			}
+			return num * mult, nil
+		}
+	}
+
+	// Plain number (bytes)
+	return strconv.ParseInt(s, 10, 64)
+}
+
```
Comment (tuxcanfly) [inline cmd/daserver/config.go]: should use ordered slice

```suggestion
// parseByteSize parses human-readable byte sizes like "2MB", "1024KB", "2097152"
func parseByteSize(s string) (int64, error) {
	s = strings.TrimSpace(strings.ToUpper(s))

	units := []struct {
		suffix string
		mult   int64
	}{
		{"GB", 1024 * 1024 * 1024},
		{"MB", 1024 * 1024},
		{"KB", 1024},
		{"B", 1},
	}

	for _, u := range units {
		if strings.HasSuffix(s, u.suffix) {
			numStr := strings.TrimSuffix(s, u.suffix)
			num, err := strconv.ParseInt(strings.TrimSpace(numStr), 10, 64)
			if err != nil {
				return 0, fmt.Errorf("invalid number in byte size: %w", err)
			}
			return num * u.mult, nil
		}
	}

	// Plain number (bytes)
	return strconv.ParseInt(s, 10, 64)
}

```

**Review by tuxcanfly — CHANGES_REQUESTED** [review]

**Review by tuxcanfly — COMMENTED** [review]
While we are here, we should clarify the changes to fallback and cache. Also document the rationale for caching get result in fallback see: https://github.com/celestiaorg/op-alt-da/pull/49#discussion_r2603084883

Also, previously `handleRedundantWrites` and `multiSourceRead` used `keccak256` of the commitment as the key to the cache/fallback. This should either be restored for compatibility or the breaking change should be documented.

See:

https://github.com/celestiaorg/op-alt-da/blob/e21b8cc4984b1f74611ce8965cbf5cabc600ebbd/fallback/s3/s3.go#L168-L173

vs 

https://github.com/celestiaorg/op-alt-da/blob/4bddf5d9b3e194f694221117dd29c87f053b261d/celestia_server.go#L286

Release checklist:

- [x] test s3 fallback including migration
- [x] note that cache was removed in release notes
- [x] test status codes for success, not found, failure
- [x] test timeouts, context deadlines and other corner cases

**Comment by Bidon15** [comment]
Regarding Release Notes for v0.10.0 after we merge this PR, here is the draft proposal: 

# Release Notes: v0.10.0

## Highlights

**v0.10.0** is a major release that simplifies configuration, improves performance with parallel blob submissions, and makes S3 as an optional fallback.

### TL;DR

- **Parallel blob submissions** - Scale throughput with multiple worker accounts
- **TOML configuration files** - Clean, version-controllable config via `--config`
- **Local keyring signing** - Direct CoreGRPC submission for reliable, token-free writes
- **S3 as an optional fallback** - S3 is now a redundancy layer, not primary cache storage as before
- **Flexible S3 credentials** - New `environment` and `iam` auth options

---

## Prerequisites

Before upgrading, ensure your Celestia infrastructure meets these minimum versions:

| Component     | Minimum Version | Required For                                   |
| ------------- | --------------- | ---------------------------------------------- |
| celestia-node | **v0.28.4**     | Reliable blob submissions, parallel workers    |
| celestia-app  | **v6.4.0**      | Parallel transaction support, sequence locking |

> **Warning:** Parallel workers will not function correctly with older versions.

---

## New Features

### TOML Configuration File Support

Run the server with a simple, readable config file:

```bash
./bin/da-server --config=config.toml
```

See `config.toml.example` for a complete template with all options documented.

### Parallel Blob Submissions

Need higher throughput? Enable parallel workers:

```toml
[celestia]
tx_worker_accounts = 4  # Use 4 worker accounts for parallel submissions
```

### Improved S3 Credential Support

S3 credentials now support three modes:

- `static` - Explicit access key and secret
- `environment` - AWS environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
- `iam` - IAM role

### Fallback Behavior

When fallback is enabled:

- **Write-through**: Blobs are written to S3 asynchronously after Celestia submission
- **Read-fallback**: If Celestia retrieval fails, attempt S3
- **Read-through**: Blobs read from Celestia are cached to S3

---

## Breaking Changes

### Local Keyring Required for Blob Submission

**Auth tokens are no longer supported for blob.Submit.** All submissions now use CoreGRPC with a local keyring:

```toml
[celestia]
core_grpc_addr = "consensus-full-mocha-4.celestia-mocha.com:9090"
keyring_path = "~/.celestia-light-mocha-4/keys"
default_key_name = "my_celes_key"
p2p_network = "mocha-4"
```

### S3 Flags Renamed to Fallback

S3 is now a fallback provider. All `--s3.*` flags have moved to `--fallback.s3.*`:

| v0.9.x                   | v0.10.0                           |
| ------------------------ | --------------------------------- |
| `--s3.credential-type`   | `--fallback.s3.credential-type`   |
| `--s3.bucket`            | `--fallback.s3.bucket`            |
| `--s3.path`              | `--fallback.s3.prefix`            |
| `--s3.endpoint`          | `--fallback.s3.endpoint`          |
| `--s3.access-key-id`     | `--fallback.s3.access-key-id`     |
| `--s3.access-key-secret` | `--fallback.s3.access-key-secret` |
| `--s3.timeout`           | `--fallback.s3.timeout`           |
| `--routing.fallback`     | `--fallback.enabled`              |
| `--routing.cache`        | _(removed)_                       |

### Default Value Changes

| Setting                  | v0.9.x                  | v0.10.0                   |
| ------------------------ | ----------------------- | ------------------------- |
| `--celestia.tls-enabled` | `true`                  | `false`                   |
| S3 timeout               | `5s`                    | `30s`                     |
| CoreGRPC address         | `http://localhost:9090` | _(required when enabled)_ |

---

## Migration Guide

### Before (v0.9.x)

```bash
./bin/da-server \
  --celestia.namespace="$NAMESPACE" \
  --celestia.server="http://localhost:26658" \
  --celestia.auth-token="$AUTH_TOKEN" \
  --celestia.tx-client.core-grpc.addr="consensus-full-mocha-4.celestia-mocha.com:9090" \
  --celestia.tx-client.keyring-path="$HOME/.celestia-light-mocha-4/keys" \
  --celestia.tx-client.key-name="my_celes_key" \
  --celestia.tx-client.p2p-network="mocha-4" \
  --s3.bucket="my-bucket" \
  --s3.credential-type="static" \
  --s3.access-key-id="$AWS_KEY" \
  --s3.access-key-secret="$AWS_SECRET" \
  --routing.fallback=true
```

### After (v0.10.0)

```bash
./bin/da-server --config=config.toml
```

**config.toml:**

```toml
[celestia]
namespace = "00000000000000000000000000000000000000000000000000000000acfe"
bridge_addr = "http://localhost:26658"
bridge_auth_token = ""
core_grpc_addr = "consensus-full-mocha-4.celestia-mocha.com:9090"
core_grpc_tls_enabled = true
keyring_path = "~/.celestia-light-mocha-4/keys"
default_key_name = "my_celes_key"
p2p_network = "mocha-4"

[fallback]
enabled = true
provider = "s3"

[fallback.s3]
bucket = "my-bucket"
region = "us-east-1"
credential_type = "environment"
```

---

## Removed

- `--routing.cache` flag
- S3 as primary storage option

---

## Documentation

- See [MIGRATION.md](./doc/MIGRATION.md) for detailed migration instructions
- See [config.toml.example](./config.toml.example) for all configuration options
- See [README.md](./README.md) for complete documentation

---

**Review by tuxcanfly — APPROVED** [review]
OK.

I think handling both keys with / without keccak hash of commitment is a bit of an overkill. We can follow up in #44 to clean this up.
